File: Soal3/Soal3.py
# Nama program : Tugas 8 UTS No. 3.py
# Nama : Gilman Nahdi Al-fathir
# NPM : 140810210060
# Tanggal buat : 23-10-2022
# Deskripsi : Matriks jadi 0 atau 1
#-------------------------------------------------

class Matriks:
    __baris = int(0)
    __kolom = int(0)
    __nilai = []

    def __init__(self, baris, kolom):
        self.__baris = baris
        self.__kolom = kolom
        self.__nilai = []
      
    def setBaris(self, baris):
        self.__baris = baris
    
    def getBaris(self):
        return self.__baris
    
    def setKolom(self, kolom):
        self.__kolom = kolom
    
    def getKolom(self):
        return self.__kolom
        
    def inputNilai(self) :
        bil = int(0)
        i = int(0)
        while (i < self.getBaris()) :
            a = []
            j = int(0)
            while (j < self.getKolom()) :
                print("Masukkan isi indeks (", i + 1, ",", j + 1, ")", end =": ")
                bil = int(input())
                a.append(bil)
                j += 1
            self.__nilai.append(a)
            i += 1
            
    def cetakMatriks(self) :
        i = int(0)
        while (i < self.getBaris()) :
            j = int(0)
            while (j < self.getKolom()) :
                print(self.__nilai[i][j],"\t", end =" ")
                j += 1
            print(" ")
            i += 1
            
    def hasilMatriks(self, A, B, C) :
        bil = int(0)
        i = int(0)
        while (i < self.getBaris()) :
            c = []
            j = int(0)
            while (j < self.getKolom()) :
                if (A.__nilai[i][j] >= B.__nilai[i][j]) :
                    bil = int(1)
                else :
                    bil = int(0)
                c.append(bil)
                j += 1
            C.__nilai.append(c)
            i += 1
    
print("Masukkan ukuran ordo matriks : ")
ko = int(input("kolom : "))
ba = int(input("baris : "))

A = Matriks(ba, ko)
B = Matriks(ba, ko)
C = Matriks(ba, ko)

print("\nInput nilai Matriks 1 : ")
A.inputNilai()

print("\nInput nilai Matriks 2 : ")
B.inputNilai()

print("\nMatriks 1 : ")
A.cetakMatriks()

print("\nMatriks 2 : ")
B.cetakMatriks()

print("\nHasil :")
C.hasilMatriks(A, B, C)
C.cetakMatriks()

File: Soal3/Soal3.cpp
/* Nama program : Tugas 8 UTS No. 3.cpp
Nama : Gilman Nahdi Al-fathir
NPM : 140810210060
Tanggal buat : 23-10-2022
Deskripsi : Matriks jadi 0 atau 1
******************************************************/

#include <iostream>

using namespace std;

class Matriks{
    private:
    int baris;
    int kolom;
    int nilai[10][10];

    public :
    Matriks(){
        baris = 0;
        kolom = 0;
    }
    Matriks(int baris, int kolom){
        this->baris = baris;
        this->kolom = kolom;
    }
    void setBaris(int baris) {
        this->baris = baris;
    }
    int getBaris(){
        return baris;
    }
    void setKolom(int kolom) {
        this->kolom = kolom;
    }
    int getKolom(){
        return kolom;
    }
    void inputNilai(){
        for (int i = 0; i < baris; i++) {
            for (int j = 0; j < kolom; j++) {
                cout << "Masukkan isi indeks (" + to_string((i + 1)) + "," + to_string((j + 1)) + ") : ";
                cin >> this->nilai[i][j];
            }
        }
    }
    void cetakMatriks(){
        for(int i = 0; i < this->baris; i++){
            for(int j=0; j < this->kolom; j++){
                cout << " " << this->nilai[i][j] << "\t";
            }
            cout<<"\n";
        }
    }
    Matriks hasilMatriks(Matriks A, Matriks B, Matriks C){
        for (int i = 0; i < baris; i++) {
            for (int j = 0; j < kolom; j++) {
                if (A.nilai[i][j] >= B.nilai[i][j]) {
                    C.nilai[i][j] = 1;
                } else {
                    C.nilai[i][j] = 0;
                }
            }
        }
        return C;
    };
};

int main(int argc, char **argv){
    int ko, ba;
    Matriks A;
    Matriks B;
    Matriks C;
    cout << "Masukkan ukuran ordo matriks : ";
    cin >> ba; cin >> ko;
    A.setBaris(ba); A.setKolom(ko);
    B.setBaris(ba); B.setKolom(ko);
    C.setBaris(ba); C.setKolom(ko);
    cout << "\nInput nilai Matriks 1 : " << endl;
    A.inputNilai();
    cout << "\nInput nilai Matriks 2 : " << endl;
    B.inputNilai();
    cout << "\nMatriks 1 : " << endl;
    A.cetakMatriks();
    cout << "\nMatriks 2 : " << endl;
    B.cetakMatriks();
    cout << "\nHasil :" << endl;
    C = C.hasilMatriks(A, B, C);
    C.cetakMatriks();
}